airbyte-integrations/connectors/source-infinicept/source_infinicept/source.py

```diff
@@ -0,0 +1,111 @@
+        self.config = config
+        self.tenant_ids = config['tenantIdList']
+        self.response_parse_name = response_parse_name
+
+    def stream_slices(self, **kwargs) -> Iterable[Optional[Mapping[str, any]]]:
+        for tenant_id in self.tenant_ids:
+            yield {"tenant_id": tenant_id}
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        return {}
+
+    def request_headers(
+            self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> Mapping[str, Any]:
+        """
+        Override to return any non-auth headers. Authentication headers will overwrite any overlapping headers returned from this method.
+        """
+        headers = {
+            'Accept': "application/json",
+            "x-AuthenticationKeyId": self.config['x-AuthenticationKeyId'],
+            "x-AuthenticationKeyValue": self.config['x-AuthenticationKeyValue']
+        }
+        return headers
+
+    def parse_response(self, response: requests.Response, stream_slice: Mapping[str, Any] = None, **kwargs) -> Iterable[Mapping]:
+        """
+        :return an iterable containing each record in the response
+        """
+        for record in response.json()[self.response_parse_name]:
+            yield self.transform(record=record, tenant_id=stream_slice["tenant_id"])
+
+    def transform(self, record: MutableMapping[str, Any], tenant_id: str)-> MutableMapping[str, Any]:
+        record['tenant_id'] = tenant_id
+        return record
+
+class Transactions(InfiniceptStream):
+    primary_key = "id"
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        tenant_id = stream_slice['tenant_id']
+        return f'/api/{tenant_id}/transactions/paginated'
+
+    def request_params(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        params = {
+            'PageSize': 1000,
+            'includeTotalCount': True
+        }
+        return params
+
+# Source
+class SourceInfinicept(AbstractSource):
+    def check_connection(self, logger, config) -> Tuple[bool, any]:
+        """
+        :param config:  the user-input config object conforming to the connector's spec.json
+        :param logger:  logger object
+        :return Tuple[bool, any]: (True, None) if the input config can be used to connect to the API successfully, (False, error) otherwise.
+        """
+        alive = True
+        error_msg = None
+
+        tenant_id = config['tenantIdList'][0]
+
+        try:
+            url = f'https://api.infinicept.com/api/{tenant_id}/fundinginstructions/paged?PageSize=1000'
```

Comment:
Do we want to use something else here, like transactions with PageSize of 1?

It's annoying that https://developer.infinicept.com/reference/diagnostics_get_heartbeat doesn't require auth, but we should probably do heartbeat first anyway, since API down and Credentials incorrect are different errors IMO.

Reply:
Agreed and this makes sense. Just a quick concern: https://developer.infinicept.com/reference/diagnostics_get_heartbeat using the Try It and later on a test file in my local, this endpoint is giving me a 404. However the other APIs are working. Is the reliability of this endpoint a concern?

Reply:
bhahaha okay, if their own API heartbeat doesn't work, you're right, we should ignore it and leave your logic as is for now.